pandas/_libs/src/ujson/python/objToJSON.c

```diff
@@ -48,10 +48,9 @@ Numeric decoder derived from from TCL library
 #include <../../../tslibs/src/datetime/np_datetime_strings.h>
 #include "datetime.h"
 
-static PyObject *type_decimal;
-
 #define NPY_JSON_BUFSIZE 32768
 
+static PyTypeObject *type_decimal;
```

Comment:
This was previously a static PyObject, but the CPython guide says you should never declare static PyObjects. Swapped over to PyTypeObject to match semantics of others in the extension:

https://docs.python.org/3/c-api/intro.html#objects-types-and-reference-counts